src/osyris/spatial/subdomain.py

```diff
@@ -0,0 +1,41 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2022 Osyris contributors (https://github.com/osyris-project/osyris)
+
+import numpy as np
+
+
+def extract_sphere(dataset, radius, origin):
+    """
+    Extract a spherical subdomain around an origin point.
+    """
+    subdomain = dataset.__class__()
```

Comment:
Why not `subdomain = Dataset()` ?

Reply:
I was just avoiding the import of `Dataset`.